### ISSUE TITLE
A few questions about AdGuard for Android

### ISSUE THREAD

**Issue by rgiotxy** [AdguardTeam/AdguardForAndroid#4174]
Hi,
I'm new to AdGuard for Android and I have few questions I'd like to ask:

1. I see an alert about having too many active DNS filters enabled on AdGuard (that it supposedly makes the filtering slow) every time I add a new one, but I see the same alert on Blokada, and at the Blokada Forum I was told not to worry about it as long as I have relatively few DNS filters and a sufficiently large (in my case 4GB) RAM memory. I was also told that, in Blokada, all active DNS filters are consolidated before being applied, so I shouldn't worry about the percentage of intersections between my active DNS filters (CPBL, OISD, hBlock and TBLP) either. Should I assume that the same goes for AdGuard (which purports to be better than Blokada), so if Blokada can handle my current volume of DNS filters and still work flawlessly smoothly, AdGuard will be able handle the same amount of DNS filters just as smoothly?
2. Do I still need the AdGuard Browser Extension if I have AdGuard for Android? I know I need it if I have Blokada, because Blokada only blocks domains while AdGuard BX can block anything (specific pages, JavaScript requisitions, CSS, even cookies and HTTP(S) requisitions), but the AdGuard for Android settings has a space for adding content-blocking filters identical to the corresponding space on the AdGuard Browser Extension settings + a separate space dedicated to Blokada-like DNS filters, so it sounds like AdGuard for Android can replace both Blokada and the AdGuard BX simultaneously.
3. **Do I need to install the AdGuard Browser Assistant?**
4. I'm told AdGuard for Android doubles as a userscript injector. Is it compatible with both GM3 and GM4? Will it inject userscripts only on web browsers, or will it work on other apps that can display webpages too? Would the settings from the AdGuard Pop-up Blocker script be applied to all apps where the script is injected or would each app require the script to be configured independently? (Incidentally, why does AdGuard need a userscript to block pop-ups? In my computer I had AdBlock Plus and the ABP Pop-ups Add-on was simply a user-friendly GUI for creating regular ABP filter rules for blocking pop-ups and pop-unders, couldn't AdGuard [which is supposedly fully compatible with ABP syntax] do that [all on its own = without a userscript] too?)
5. If AdGuard functions as a userscript injector, can I uninstall Tampermonkey/ViolentMonkey from Firefox?
6. Blokada displays all domains passing through it and whether or not they have been blocked by it, so the user can manually identify domains that are passing through but shouldn't be or vice-versa. Does AdGuard for Android do that? (Incidentally, ABP also has a Blockable Elements Console that shows every element in the page and whether it's being blocked or not so you can manually identify elements that should be blocked but aren't being or vice versa, any chance AdGuard can ever implement that?)
7. Where do I post a suggestion that AdGuard should have a native option to select which parts of the AdGuard Annoyances list you want enabled and that it should include a clickbait filters sub-list? (Although, now that I think about it… when I was configuring AdGuard, I was explicitly asked if I wanted to keep social media widgets and cookie policy notifications and I said yes, as those are my main reasons for not subscribing to the AdGuard Annoyances list as a whole and instead subscribing to sub-lists one by one — does the fact that I opted out of blocking those elements when AdGuard asked me to mean that they wouldn't be blocked if I opted into AdGuard Annoyances as a whole list instead?)
8. Is there a way to make AdGuard displays dates in the standard UN format (DD MM YYYY) rather than the USA-only format (MM DD YYYY)?

Thank you for your time!

PS: can someone please fix the labels on my previous thread (#4137) here? The correct tags should be "BUG REPORT" and "Resolution: Done" (since I did find a solution, as noted in the final post).

**Comment by Birbber** [comment]
@rgiotxy Hi! I'll try to answer all of your questions.

1. You mean `DNS filter is too large` message?
This is a leftover from the previous DnsProxy which kept all filters in memory. The new one is much more optimized and does not load filters into memory, but reads them from a file, so you don't need to worry about this.
2. There's no need in AdGuard Browser Extension, since AdGuard for Android has a big coverage. 
3. You don't need to install the AdGuard Browser Assistant, because you'll already have one in android's status bar.
4. We do support Greasemonkey API (https://wiki.greasespot.net/Greasemonkey_Manual:API). And yes, AdGuard for Android injects into embedded webviews.
5. Yes, you can install scripts as extensions (but it's available for paid users only), so there's no point in using Tampermonkey/ViolentMonkey I guess.
6. There is a filtering log in AdGuard for Android. There you can see blocked(red color), bypassed(no color), allowed(green color) and modified(yellow color) requests. You can block/unblock requests or simply view info of any request. 
7. We expect this feature in v4.0.
8. If it is about the date on the main screen -- it is formatted according to the selected language. If you choose US you get `MM DD YYYY`. Right now we don't separate US English from any other.

**Comment by rgiotxy** [comment]
@Birbber THANK YOU × 8!!!

Re: the date format, my mobile's system language is UK English, so e.g. the Android status bar / notifications pane displays "Wed, 25 May" rather than "Wed, May 25", but the AdGuard home page reads "Since May 21, 2022" instead of "Since 21 May, 2022" =(

**Comment by Birbber** [comment]
@rgiotxy 

> the date format, my mobile's system language is UK English, so e.g. the Android status bar / notifications pane displays "Wed, 25 May" rather than "Wed, May 25", but the AdGuard home page reads "Since May 21, 2022" instead of "Since 21 May, 2022" =(

Sadly we have US format for now. Maybe it will be changed later.

> AdGuard Pop-Up Blocker and other userscripts that require configuration by accessing a specific webpage, can this webpage be accessed by any browser and the resulting configurations will be applied to all apps where the script is injected?

Do you mean userscripts that add some button to the page in order to go to the settings of this userscript? AdGuard Popup Blocker doesn't have such button. But it should work both in browsers and apps.

> my other thread, can someone please update the labels? the solution was found, but the thread is still labelled "won't fix"

It means we won't fix it on our end. So even though you've found the solution it doesn't mean the devs will change anything in the code. That's why the tag says "Won't fix"

**Comment by rgiotxy** [comment]
@Birbber  
> Sadly we have US format for now. Maybe it will be changed later.
I see. 

> Do you mean userscripts that add some button to the page in order to go to the settings of this userscript? 

AdGuard Popup Blocker doesn't have such button. But it should work both in browsers and apps.
I mean https://popupblocker.adguard.com/options.html
All my settings disappeared now that I'm using AdGuard for Android. I'll try to see if they return if I disable AfA and, if so, I'll try to export them via copy and paste. But I'm wondering if doing so will only affect the browser where I accessed the page. I suppose I can figure that out on my end, though, by installing another browser and accessing it from there.

>  

Does AdGuard for Android have an option for creating automatic cosmetic filters by physically clicking on the element we wish to hide, like the AdGuard Browser Extension does?

> It means we won't fix it on our end. So even though you've found the solution it doesn't mean the devs will change anything in the code. That's why the tag says "Won't fix"

Oh, I see. It's true, the problem was on the user's end and must be resolved on the user's end via the Android Settings. I just hope other people who have the same problem don't misunderstand "won't fix" to mean "doesn't have a solution", like I probably would have. Maybe you guys could post a warning on the download website, or an instruction in the FAQ, or something like that? Something that clearly states AdGuard can't create a VPN profile if the user has a pre-existing always on VPN profile and instructs the user to simply remove the other VPN profiles' "always on" privileges (no need to uninstall them) so AdGuard can create its own? [The current version of the FAQ claims this is always a firmware problem and that it cannot be solved on the user's side](https://kb.adguard.com/en/android/faq#supp), which in my case wasn't true at all.
